app/controllers/clearance/passwords_controller.rb

```diff
@@ -96,12 +101,27 @@ def forbid_missing_token
   end
 
   def forbid_non_existent_user
-    unless find_user_by_id_and_confirmation_token
+    unless find_user_from_password_reset
       flash_failure_when_forbidden
       render template: 'passwords/new'
     end
   end
 
+  def find_user_from_password_reset
+    if matching_password_reset = find_password_reset_by_user_id_and_token
+      matching_password_reset.user
+    end
+  end
+
+  def forbid_expired_password_reset
```

Comment:
would this fit as a validation on password reset instead? validate it's not expired, call `valid?` as part of the theoretical `complete` method, perhaps?